Code/Framework/AzCore/Tests/Name/NameTests.cpp

```diff
@@ -897,5 +898,89 @@ namespace UnitTest
 
         EXPECT_LT(nameTime, stringTime);
     }
+
+    // Fixture for validating that multiple NameDictionarys can exist at once.
+    class MultiNameDictionaryFixture
+        : public ScopedAllocatorSetupFixture
+    {
+    public:
+        MultiNameDictionaryFixture()
+        {
+            m_nameDictionary1 = AZStd::make_unique<AZ::NameDictionary>();
+            m_nameDictionary2 = AZStd::make_unique<AZ::NameDictionary>();
+        }
+
+        ~MultiNameDictionaryFixture()
+        {
+            m_nameDictionary2.reset();
+            m_nameDictionary1.reset();
+        }
+
+    protected:
+        AZStd::unique_ptr<AZ::NameDictionary> m_nameDictionary1;
+        AZStd::unique_ptr<AZ::NameDictionary> m_nameDictionary2;
+    };
+
+    TEST_F(MultiNameDictionaryFixture, MultipleDictionaries_Contains_DifferentNameInstances)
+    {
+        // Dictionary 1 will contain the names of "Name1" and "Dictionary1NameOnly"
+        AZ::Name name1FromDict1 = m_nameDictionary1->MakeName("Name1");
+        AZ::Name dict1OnlyName = m_nameDictionary1->MakeName("Dictionary1NameOnly");
+
+        // Dictionary 2 will contain the names of "Name1" and "Dictionary2NameOnly"
+        AZ::Name name1FromDict2 = m_nameDictionary2->MakeName("Name1");
+        AZ::Name dict2OnlyName = m_nameDictionary2->MakeName("Dictionary2NameOnly");
+
+        // First dictionary EXPECTS
+        EXPECT_FALSE(m_nameDictionary1->FindName(name1FromDict1.GetHash()).IsEmpty());
+        EXPECT_FALSE(m_nameDictionary1->FindName(dict1OnlyName.GetHash()).IsEmpty());
+        // The first dictionary should NOT contain Dictionary2NameOnly
+        EXPECT_TRUE(m_nameDictionary1->FindName(dict2OnlyName.GetHash()).IsEmpty());
+
+        // Second dictionary EXPECTS
+        EXPECT_FALSE(m_nameDictionary2->FindName(name1FromDict1.GetHash()).IsEmpty());
+        EXPECT_FALSE(m_nameDictionary2->FindName(dict2OnlyName.GetHash()).IsEmpty());
+        // The second dictionary should NOT contain Dictionary1NameOnly
+        EXPECT_TRUE(m_nameDictionary2->FindName(dict1OnlyName.GetHash()).IsEmpty());
+
+        // reset the name1FromDict1 variable
+        // Only the reference to "Name1" entry in first dictionary should be removed
+        const AZ::Name::Hash name1Hash = name1FromDict1.GetHash();
+        name1FromDict1 = AZ::Name{};
+
+        EXPECT_TRUE(m_nameDictionary1->FindName(name1Hash).IsEmpty());
+        EXPECT_FALSE(m_nameDictionary2->FindName(name1FromDict2.GetHash()).IsEmpty());
+    }
+
+    TEST_F(MultiNameDictionaryFixture, NameLiteral_IsOnlyLinkedToSingleNameDictionary)
+    {
+        // When run with --gtest_repeat=2 the literal is not relinked to the NameDictionary
```

Comment:
Yeah, this makes testing a little funky, only really works with lazy re-linking, but at least in terms of practical use it should be OK.